### PR TITLE
[v1.14.x] Fix Transformation Logging Global Setting

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description
 - Backport #8393 to v1.14.x

### CODE REVIEW

File: changelog/v1.14.9/transformation-logging-global-settings-fix.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+changelog:
+  - type: FIX
+    issuelink: https://github.com/solo-io/gloo/issues/8266
+    resolvesIssue: false
```
Comment (ashishb-solo) [inline changelog/v1.14.9/transformation-logging-global-settings-fix.yaml:4]: should this be true?

Comment (ben-taussig-solo) [reply changelog/v1.14.9/transformation-logging-global-settings-fix.yaml:4]: I don't think this is really resolved until we bump the Gloo OSS dependency in solo-projects on main and v1.14.x, but I guess it's kind of ambiguous

Comment (ben-taussig-solo) [reply changelog/v1.14.9/transformation-logging-global-settings-fix.yaml:4]: I plan on updating the issue manually with links to enterprise releases that contain the fix, so I'd rather not close the issue when this PR merges

**Review by jenshu — APPROVED** [review]

File: changelog/v1.14.9/transformation-logging-global-settings-fix.yaml (hunk 1 above)
Comment (ashishb-solo) [reply changelog/v1.14.9/transformation-logging-global-settings-fix.yaml:4]: sounds good to me!

**Review by ashishb-solo — APPROVED** [review]

**Pull request merged**